### PR TITLE
minimal HTTP content routing client

### PR DESCRIPTION
**Pull request by willscott**

### CODE REVIEW

File: contentrouting/interface.go [hunk 1]
```diff
@@ -3,23 +3,51 @@ package contentrouting
...
 )
 
 // The Routing interface is an evolution of the ContentRouting interface
 // found in [libp2p](https://github.com/libp2p/go-libp2p-core/blob/master/routing/routing.go#L26).
 //
 // Routing focuses only on the retrieval half of the interface: how do
 // I located content given a content identifier?
-interface Routing {
+type Routing interface {
 	FindProvidersAsync(context.Context, cid.Cid) <-chan RoutingRecord
```
Comment (hannahhoward) [inline contentrouting/interface.go:16]: Just to be safe, I'd add a varargs in the parameters.... just call it a ContentRoutingOption with a blank interface for now?

Given we are returning a channel, we should specify what happens if the consumer fails to consume it. (i.e. does our system get stuck? Can we keep processing effectively?)

**Review by hannahhoward — APPROVED** [review]
I think LGTM? But consider (non-blocking) var args in the routing interface

**Comment by rvagg** [comment]
out of interest, what kinds of options might one expect to be passed here? or are we defining a consistent interface that's going to be copied for other transports and therefore expect a lot of option-chatter?

**Comment by willscott** [comment]
This is a good question. I imagine most transport-specific options would be preferably specified in the constructor to the transport, rather than at the content-routing fetch time.

I think we could imagine in the scope of one call to ask for:
* hints about the selector that is triggering the request, or the parent cid that this one is downstream of
* relative priority of the request (e.g. 'don't open new network requests for this one')

**Pull request merged**